core/templates/dev/head/i18n.js

```diff
@@ -45,20 +45,26 @@ oppia.constant('DEFAULT_TRANSLATIONS', {
 });
 
 oppia.controller('I18nFooter', [
-    '$rootScope', '$scope', '$translate',
-    function($rootScope, $scope, $translate) {
+    '$http', '$rootScope', '$scope', '$translate',
+    function($http, $rootScope, $scope, $translate) {
   // Changes the language of the translations.
+  var _PREFERENCES_DATA_URL = '/preferenceshandler/data';
+  var _SAVE_SITE_LANGUAGE_URL = '/save_site_language';
   $scope.supportedSiteLanguages = GLOBALS.SUPPORTED_SITE_LANGUAGES;
+  if (GLOBALS.userIsLoggedIn) {
+    $http.get(_PREFERENCES_DATA_URL).success(function(data) {
+      $translate.use(data.preferred_site_language_code);
```

Comment:
I _think_ that might be fine, but I'm a little bit confused, sorry. Why do you need to know that they "just logged in" -- i.e., why is the "just" important?

If it's not, and you just care that they are logged in, then you could read off their preference from the user_settings instance that's already retrieved anyway in BaseHandler: https://github.com/oppia/oppia/blob/9ebbc35baa4446cc03828ffa52e8a589c2a531ba/core/controllers/base.py#L296

Then you can set it in the template: https://github.com/oppia/oppia/blob/9ebbc35baa4446cc03828ffa52e8a589c2a531ba/core/controllers/base.py#L393

Does that help?